### PR TITLE
[Ruby 2.5] Add specs for String#each_grapheme_cluster and String#grapheme_clusters

### PR DESCRIPTION
**Pull request by nilsding**
Something for #565.  Was inspired by the specs for `#chars` and `#each_chars` for this one, and I also reused the shared specs for those as they behave mostly the same.  Suggestions welcome! 😄

### CODE REVIEW

File: core/string/shared/grapheme_clusters.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+describe :string_grapheme_clusters, shared: true do
+  it "passes each char in self to the given block" do
+    a = []
+    # test string: abc[rainbow flag emoji][paw prints]
+    "ab\u{1f3f3}\u{fe0f}\u{200d}\u{1f308}\u{1F43E}".send(@method) { |c| a << c }
+    a.should == ['a', 'b', "\u{1f3f3}\u{fe0f}\u{200d}\u{1f308}", "\u{1F43E}"]
+  end
+
+  it "returns self" do
```
Comment (nilsding) [inline core/string/shared/grapheme_clusters.rb]: Not sure if it makes sense to have that one in here as it's technically covered by the ones for `:string_chars`.

File: core/string/shared/grapheme_clusters.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative '../../../spec_helper'
+require_relative '../fixtures/classes'
+
+describe :string_grapheme_clusters, shared: true do
+  it "passes each char in self to the given block" do
```
Comment (eregon) [inline core/string/shared/grapheme_clusters.rb]: Each grapheme cluster ;)

File: core/string/shared/grapheme_clusters.rb (hunk 1 above)
Comment (eregon) [reply core/string/shared/grapheme_clusters.rb]: Seems fine and it doesn't hurt :)

File: core/string/grapheme_clusters_spec.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require_relative 'shared/chars'
+require_relative 'shared/grapheme_clusters'
+
+ruby_version_is "2.5" do
+  describe "String#grapheme_clusters" do
```
Comment (eregon) [inline core/string/grapheme_clusters_spec.rb]: Could you add a spec that this method returns an Array if called without a block?

File: core/string/shared/grapheme_clusters.rb (hunk 2 above)
Comment (nilsding) [reply core/string/shared/grapheme_clusters.rb]: Oh, right 😅

File: core/string/grapheme_clusters_spec.rb (hunk 3 above)
Comment (nilsding) [reply core/string/grapheme_clusters_spec.rb]: Sure thing!

**Comment by eregon** [comment]
Merged, thank you!

**Pull request merged**